src/group_theory/group_action/prod.lean

```diff
@@ -41,6 +41,20 @@ variables [has_smul M α] [has_smul M β] [has_smul N α] [has_smul N β] (a : M
 @[to_additive] theorem smul_def (a : M) (x : α × β) : a • x = (a • x.1, a • x.2) := rfl
 @[simp, to_additive] theorem smul_swap : (a • x).swap = a • x.swap := rfl
 
+@[to_additive has_smul]
+instance has_pow {γ : Type*} [has_pow α γ] [has_pow β γ] : has_pow (α × β) γ :=
+{ pow := λ p c, (p.1 ^ c, p.2 ^ c) }
```

Comment:
I recomment using `E` instead of `γ` (mnemonic: exponent); then you can move all the `has_pow` variables up to line 34 and change line 28 to

```
variables {M N P α β E : Type*}
```

Reply:
renamed